private/syntax-replacement.rkt

```diff
@@ -12,11 +12,15 @@
  (contract-out
   [syntax-replacement? predicate/c]
   [syntax-replacement
-   (-> #:original-syntax (and/c syntax? syntax-original?) #:new-syntax syntax? syntax-replacement?)]
+   (-> #:original-syntax (and/c syntax? syntax-original?)
+       #:new-syntax syntax?
+       #:introduction-scope (->* (syntax?) ((or/c 'flip 'add 'remove)) syntax?)
```

Comment:
I really wish there was a `scope?` data type exposed from the standard library that wrapped these introducers so we didn't need complex contracts just to create and pass around scopes. Maybe someday someone will get around to adding a `racket/scope` module to the standard library or something.